snr.py

```diff
@@ -0,0 +1,287 @@
+                  gain=u.electron / u.adu,
+                  ad_err=u.adu / u.pixel)
+def snr(counts,
+        npix=1 * u.pixel,
+        n_background=np.inf * u.pixel,
+        background=0 * u.electron / u.pixel,
+        darkcurrent=0 * u.electron / u.pixel,
+        readnoise=0 * u.electron / u.pixel,
+        gain=1 * u.electron / u.adu,
+        ad_err=np.sqrt(0.289) * u.adu / u.pixel):
+    """
+    A function to calculate the signal to noise ratio (SNR) of an
+    astronomical observation with a given number of counts (from
+    "Handbook of CCD Astronomy", Steve Howell, 2000, pg 55).
+
+    Parameters
+    ----------
+    counts : `~astropy.units.Quantity`
+        Total number of counts in an arbitrary exposure time with units of
+        electrons.
+    npix : `~astropy.units.Quantity`, optional
+        Number of pixels under consideration for the signal with units of
+        pixels. Default is 1 * astropy.units.pixel.
+    n_background : `~astropy.units.Quantity`, optional
+        Number of pixels used in the background estimation with units of
+        pixels. Default is set to np.inf * astropy.units.pixel
+        such that there is no contribution of error due to background
+        estimation. This assumes that n_background will be >> npix.
+    background : `~astropy.units.Quantity`, optional
+        Total photons per pixel due to the background/sky with units of
+        electrons/pixel.
+        Default is 0 * astropy.units.electron / astropy.units.pixel.
+    darkcurrent : `~astropy.units.Quantity`, optional
+        Total electrons per pixel due to the dark current with units
+        of electrons/pixel.
+        Default is 0 * astropy.units.electron / astropy.units.pixel.
+    readnoise : `~astropy.units.Quantity`, optional
+        Electrons per pixel from the read noise with units of electrons/pixel.
+        Default is 0 * astropy.units.electron / astropy.units.pixel.
+    gain : `~astropy.units.Quantity`, optional
+        Gain of the CCD with units of electrons/ADU. Default is
+        1 * astropy.units.electron / astropy.units.adu such that the
+        contribution to the error due to the gain is assumed to be small.
+    ad_err : `~astropy.units.Quantity`, optional
+        An estimate of the 1 sigma error within the A/D converter with units of
+        adu/pixel. Default is set to
+        0.289 * astropy.units.adu / astropy.units.pixel
+        (Merline & Howell, 1995).
+
+    Returns
+    -------
+    sn : `~astropy.units.Quantity`
+        The signal to noise ratio of the given observation in untis of
+        sqrt(electrons).
+    """
+    readnoise = get_shotnoise(readnoise)
+    gain_err = get_shotnoise(gain * ad_err)
+
+    pixel_terms = npix * (1 + npix / n_background)
+    detector_noise = (background + darkcurrent +
+                      readnoise ** 2 + gain_err ** 2)
+
+    sn = counts / np.sqrt(counts + pixel_terms * detector_noise)
+
+    return sn
+
+
+def get_shotnoise(detector_property):
```

Comment:
An organizational point: in Python, we often differentiate between "public" functions and "private" functions, i.e. the ones you want people to use = public, and the nitty-gritty utility functions that are meant to be guts of public functions = private. I think this function and `t_with_small_errs` are good examples of private functions, which I know you already recognize because your docstrings are sparser than normal for these functions. 

The way we denote private functions typically is by starting the function with a single underscore, i.e.: `_get_shotnoise` and `_t_with_small_errors`.

Reply:
Ohh, I like that. I learn something new every day with you guys :)